src/uri/include/sourcemeta/jsontoolkit/uri.h

```diff
@@ -318,6 +323,15 @@ class SOURCEMETA_JSONTOOLKIT_URI_EXPORT URI {
   // points to fragments of it.
   // We keep this as const as this class is immutable
   std::string data;
+
+  std::optional<std::string> path_;
```

Comment:
A bit of a nit pick, but would it make sense to re-order these members to reflect the actual ordering in a URI? For example, start with the scheme, then userinfo, etc? I think it'd make it easier to read

Reply:
Yeah I should do the same with the parse method.